graphql/resolver.go

```diff
@@ -68,5 +70,32 @@ func New(apiURL string) Config {
 		}
 		return nil, ResourceNotFound.Send(ctx, "Could not find project")
 	}
+	c.Directives.RestrictProjectAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver) (res interface{}, err error) {
+		user := gimlet.GetUser(ctx)
+		if user == nil {
+			return nil, Forbidden.Send(ctx, "user not logged in")
+		}
+
+		projectRef, isProjectRef := obj.(*restModel.APIProjectRef)
+		if !isProjectRef {
+			return nil, InternalServerError.Send(ctx, "project not valid")
+		}
+
+		projectId := utility.FromStringPtr(projectRef.Id)
+		if projectId == "" {
+			return nil, ResourceNotFound.Send(ctx, "project not specified")
+		}
+
+		opts := gimlet.PermissionOpts{
+			Resource:      projectId,
+			ResourceType:  evergreen.ProjectResourceType,
+			Permission:    evergreen.PermissionProjectSettings,
+			RequiredLevel: evergreen.ProjectSettingsView.Value,
+		}
+		if user.HasPermission(opts) {
```

Comment:
https://github.com/evergreen-ci/evergreen/blob/d1f28a88613c3789e22ad4b1ddd5f2c3935b284c/graphql/util.go#L775 

Is this method applicable here? If not do you think it would make sense to adapt it so we have one way of doing this pattern?

Reply:
I refactored it so that the directives all follow the same structure, let me know what you think!